### PR TITLE
more fixes to device changes

### PR DESCRIPTION
**Pull request by snylonue**
- create CpalOutputStream on demand
- patch cpal to respect windows device selection

for testing: https://github.com/snylonue/annix/actions/runs/7308694756

### CODE REVIEW

**Comment by Yesterday17** [comment]
I'm going to review this PR tonight.

**Comment by snylonue** [comment]
related issue: https://github.com/erikas-taroza/simple_audio/issues/48

File: anni-playback/src/decoder/decoder.rs [hunk 1]
```diff
@@ -190,28 +190,10 @@ impl Decoder {
                 // and pause playback. Once the user is ready, they can start
                 // playback themselves.
                 InternalPlayerEvent::DeviceChanged => {
+                    log::info!("device changed");
```
Comment (Yesterday17) [inline anni-playback/src/decoder/decoder.rs:193]: info -> debug?

File: anni-playback/src/cpal_output.rs [hunk 2]
```diff
@@ -165,6 +165,8 @@ impl CpalOutputStream {
             .default_output_device()
             .context("Failed to get default output device.")?;
 
+        log::info!("default device: {:?}", device.name());
```
Comment (Yesterday17) [inline anni-playback/src/cpal_output.rs:168]: info -> debug?

File: anni-playback/src/decoder/decoder.rs [hunk 3]
```diff
@@ -356,9 +340,9 @@ impl Decoder {
         }
 
         // If there is a preloaded file, then swap it with the current playback.
-        if let Some((playback, cpal_output)) = self.preload_playback.take() {
+        if let Some(playback) = self.preload_playback.take() {
             self.playback = Some(playback);
-            self.cpal_output = Some(cpal_output);
+            // self.cpal_output = Some(cpal_output);
```
Comment (Yesterday17) [inline anni-playback/src/decoder/decoder.rs:345]: 这个直接删掉吧

**Comment by snylonue** [comment]
By the way, is there any glitches during playback?

**Comment by Yesterday17** [comment]
> By the way, is there any glitches during playback?

It plays seamlessly with no glitch heard.

**Review by Yesterday17 — APPROVED** [review]

**Pull request merged**